src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -663,6 +786,28 @@ def _get_list_runs_message(base_message, task_name=None, image=None):
     return "{}.".format(base_message)
 
 
+def _build_identities_info(cmd, identities, is_remove=None):
+    IdentityProperties, UserIdentityProperties, ResourceIdentityType = cmd.get_models(
+        'IdentityProperties', 'UserIdentityProperties', 'ResourceIdentityType')
+    identities = identities or []
+    identity_types = []
+    if IDENTITY_GLOBAL_REMOVE in identities:
+        return IdentityProperties(type=ResourceIdentityType.none.value)
+    if not identities or IDENTITY_LOCAL_ID in identities:
+        identity_types.append(ResourceIdentityType.system_assigned.value)
+    external_identities = [x for x in identities if x != IDENTITY_LOCAL_ID]
+    if external_identities:
+        identity_types.append(ResourceIdentityType.user_assigned.value)
+    identity_types = ', '.join(identity_types)
```

Comment:
Do you change the type of IdentityProperties from array to string here?

Reply:
yep its an Enum in RP.

Reply:
Yes, this is just for the `SystemAssigned, UserAssigned` case. This just checks if the user is assigning both `SystemAssigned` and `UserAssigned` identities, then the type is `SystemAssigned, UserAssigned`. The identity `Type` is a string. When there is one identity type in the list, no join is performed and it is rendered as a string during `type=identity_types`